src/main/daml/Daml/Finance/Claims/Lifecycle/Rule.daml

```diff
@@ -153,9 +153,9 @@ tryCreateNewInstrument actor prevElections eventTime electionData oldKey
         instrumentCid <- BaseInstrument.exerciseInterfaceByKey @DynamicInstrument.I oldKey
           actor DynamicInstrument.CreateNewVersion with
             lastEventTimestamp = eventTime
-            prevElections = case electionData of
-              None -> prevElections
-              Some (electorIsOwner, claimTag) -> prevElections ++ [newElection]
+            prevEvents = case electionData of
+              None -> prevEvents ++ [timeEvent eventTime]
```

Comment:
If I understand it correctly, we are always adding the current `eventTime` to the list of previous events. 

This might be undesirable, as for a bond we would end up lifecycling many times (once for each coupon date) whereas that is not required for fast-forwarding

Reply:
> This might be undesirable, as for a bond we would end up lifecycling many times (once for each coupon date) whereas that is not required for fast-forwarding

Actually, for instruments without any Elections or other path dependencies, we do not populate prevEvents. For example, for a fixed coupon bond:

```
    interface instance DynamicInstrument.I for Instrument where
      view = DynamicInstrument.View with lifecycler = issuer; lastEventTimestamp; prevEvents = []
      createNewVersion DynamicInstrument.CreateNewVersion{version; lastEventTimestamp} = do
        cid <- create this with version; lastEventTimestamp
        BaseInstrument.createReference issuer $ toInterfaceContractId cid
        pure $ toInterfaceContractId cid
```

However, your observation holds true for instruments like callable bonds, where some dates must be stored (call dates) whereas others are not required (if the bond has some regular coupon dates without callability). I was considering that during implementation, but decided it was quite minor. What do you think?

Reply:
> I think it's not the end of the world, as every time we would be lifecycling on a smaller tree, but still inefficient. I wonder if there is a way to make the list of prevEvents a per-instrument decision

The per-instrument decision we already have, but we could make it even more efficient by also doing it on a per-event basis (for those instruments were events are generally stored). That would require us to determine, during the lifecycling of an event, whether it has a path dependant impact on future events. Do you see an easy way there?

Reply:
I opened an issue to track this: https://github.com/digital-asset/daml-finance/issues/886